scripts/tasks/task-loadfw.py

```diff
@@ -122,31 +131,85 @@ def make_sysex_messages(binary, handshake):
     return messages
 
 
-def load_fw(port, handshake, file, delay_ms=2, output_to_file=False):
+def load_fw(output, handshake, file, delay_ms=2, output_to_file=False):
     with open(file, "rb") as f:
         binary = f.read()
 
     sysex_data = make_sysex_messages(binary, handshake)
 
     if output_to_file:
-        with open(port, "wb") as f:
+        with open(output, "wb") as f:
             text = f"Writing SysEx File '{f.name}': "
             for msg in util.progressbar(sysex_data, text):
                 f.write(msg)
 
     else:
-        midiout = rtmidi.MidiOut()
-        midiout.open_port(int(port))
-        with midiout:
+        with output:
             for msg in util.progressbar(sysex_data, "Firmware Upload: "):
-                midiout.send_message(msg)
+                output.send_message(msg)
                 time.sleep(0.001 * delay_ms)
 
 
+def try_read_hex_key():
+    name = ".deluge_hex_key"
+    if os.path.isfile(".deluge_hex_key"):
+        with open(name, "r") as f:
+            return f.read().strip()
+    else:
```

Comment:
a `try/except` block night work better here, as generally filesystems are more of a "ask forgiveness than permission" sort of entity (consider e.g. a file that exists but it's not readable).

Reply:
Also consider the file might exist while checking but not while running if this is ran in parallel with a git clean or something similar